docs/tutorials/ml_inference/semantic_search/optimize_vector_search_with_cohere_compressed_embedding.md

```diff
@@ -0,0 +1,536 @@
+- [k-NN index](https://opensearch.org/docs/latest/search-plugins/knn/index/) and [k-NN byte vector](https://opensearch.org/docs/latest/field-types/supported-field-types/knn-vector/#byte-vectors)
+
+Note: Replace the placeholders that start with `your_` with your own values.
+
+# Steps
+
+## 1. Create Embedding Model
+
+First, we'll set up a connector to Amazon Bedrock for accessing Cohere's embedding model.
+
+
+### 1.1 Create Connector
+Create a connector by following this [blueprint](https://github.com/opensearch-project/ml-commons/blob/main/docs/remote_inference_blueprints/bedrock_connector_cohere_cohere.embed-multilingual-v3_blueprint.md).
+
+Note: We don't need  pre/post process function in this tutorial as we are going to use ML inference processor.
+
+`"embedding_types": ["int8"]` specifies that we want 8-bit integer quantized embeddings from the Cohere model.
+This meaans embeddings are compressed from 32-bit floats to 8-bit integers.
+It reduces storage space and improves computation speed.
+There's a minor trade-off in precision, but it's usually negligible for search tasks.
+It's compatible with OpenSearch's KNN index, which supports byte vectors.
+
+You can find more details about model parameters on [Cohere docment](https://docs.cohere.com/v2/docs/embeddings).
+
+```
+POST _plugins/_ml/connectors/_create
+{
+  "name": "Amazon Bedrock Connector: Cohere embed-multilingual-v3",
+  "description": "Test connector for Amazon Bedrock Cohere embed-multilingual-v3",
+  "version": 1,
+  "protocol": "aws_sigv4",
+  "credential": {
+    "access_key": "your_aws_access_key",
+    "secret_key": "your_aws_secret_key",
+    "session_token": "your_aws_session_token"
+  },
+  "parameters": {
+    "region": "your_aws_region",
+    "service_name": "bedrock",
+    "truncate": "END",
+    "input_type": "search_document",
+    "model": "cohere.embed-multilingual-v3",
+    "embedding_types": ["int8"]
+  },
+  "actions": [
+    {
+      "action_type": "predict",
+      "method": "POST",
+      "headers": {
+        "x-amz-content-sha256": "required",
+        "content-type": "application/json"
+      },
+      "url": "https://bedrock-runtime.${parameters.region}.amazonaws.com/model/${parameters.model}/invoke",
+      "request_body": "{ \"texts\": ${parameters.texts}, \"truncate\": \"${parameters.truncate}\", \"input_type\": \"${parameters.input_type}\", \"embedding_types\":  ${parameters.embedding_types} }"
+
+    }
+  ]
+}
+```
+Sample response
+```
+{
+  "connector_id": "AOP0OZUB3JwAtE25PST0"
+}
+```
+
+### 1.2 Create Model
+Now, let's create a model using the connector we just set up:
+
+Note: The `interface` parameter is optional. If you don't need an interface, you can simply set it as an empty object: `"interface": {}`. For more information about model interfaces, see the [model interface documentation](https://opensearch.org/docs/latest/ml-commons-plugin/api/model-apis/register-model/#the-interface-parameter).
```

Comment:
A little bit confusing, looks like the `interface` is not optional, as even if it is not needed, we still have to configure an empty `"interface": {}`, is it correct?

Reply:
Our code will create interface automatically for some model. This model included. That means if not provide `"interface": {}`, our [code](https://github.com/opensearch-project/ml-commons/blob/2.19/common/src/main/java/org/opensearch/ml/common/utils/ModelInterfaceUtils.java#L834) will automatically create interface. If model not included in our code, the interface will be empty. Always adding `"interface": {}` make things easier. 
To avoid duplicate explaining here, suggest user check details on the model interface document .